docs/proposals/failover/application-failover.md

```diff
@@ -0,0 +1,566 @@
+
+    // HealthyState refers to the healthy status reported by the Karmada resource
+    // interpreter.
+    // Valid options are "Unhealthy" and "Unknown".
+    // +optional
+    HealthyState []string `json:"healthyState,omitempty"`
+}
+```
+
+For example:
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx-propagation
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+    - apiVersion: apps/v1
+      kind: StatefulSet
+      name: mysql
+  failover:
+    application:
+      initiateGates:
+        delaySeconds: 5000
+      activeGates:
+        healthyState: 
+        - Unhealthy
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member1
+        - member2
+```
+
+#### ResourceBinding API change
+
+```golang
+type ResourceBindingSpec struct {
+    ...
+    // EvictedClusters represents the evicted clusters.
+    // +optional
+    EvictedClusters []EvictedCluster `json:"evictedClusters,omitempty"`
+
+    // FailoverBehavior represents the failover config for the referencing resource.
+    // It inherits directly from the associated PropagationPolicy(or ClusterPropagationPolicy).
+    // +optional
+    FailoverBehavior *policyv1alpha1.FailoverBehavior `json:"failoverBehavior,omitempty"`
+    ...
+}
+
+// AggregatedStatusItem represents status of the resource running in a member cluster.
+type AggregatedStatusItem struct {
+    // ClusterName represents the member cluster name which the resource deployed on.
+    // +required
+    ClusterName string `json:"clusterName"`
+    
+    // Status reflects running status of current manifest.
+    // +kubebuilder:pruning:PreserveUnknownFields
+    // +optional
+    Status *runtime.RawExtension `json:"status,omitempty"`
+    // Applied represents if the resource referencing by ResourceBinding or ClusterResourceBinding
+    // is successfully applied on the cluster.
+    // +optional
+    Applied bool `json:"applied,omitempty"`
+    
+    // AppliedMessage is a human readable message indicating details about the applied status.
+    // This is usually holds the error message in case of apply failed.
+    // +optional
+    AppliedMessage string `json:"appliedMessage,omitempty"`
+    
+    // Health represents the healthy state of the current resource.
+    // There maybe different rules for different resources to achieve health status.
+    // +kubebuilder:validation:Enum=Healthy;Unhealthy;Unknown
+    // +optional
+    Health ResourceHealth `json:"health,omitempty"`
+
+    // CreationTimestamp is a timestamp representing the server time when this object was
+    // created.
+    // Clients should not set this value to avoid the time inconsistency issue.
+    // It is represented in RFC3339 form(like '2021-04-25T10:02:10Z') and is in UTC.
+    //
+    // Populated by the system. Read-only.
+    // +optional
+    CreationTimestamp metav1.Time `json:"creationTimestamp,omitempty"`
```

Comment:
I can't get the meaning and how it works from this comment.
Could you please explain more?